### PR TITLE
Always include a citation

### PR DESCRIPTION
**Pull request by maelle**
Instead of #1416

Some notes

* The authors layout and `build_authors()` function would be removed.
* I have noticed the title of the author/author citation page isn't translated yet.
* I think the citation should come after the authors.

### CODE REVIEW

File: R/build-home-citation.R [hunk 1]
```diff
@@ -31,39 +31,55 @@ read_citation <- function(path = ".") {
 data_home_sidebar_citation <- function(pkg = ".") {
   pkg <- as_pkgdown(pkg)
 
-  if (!has_citation(pkg$src_path)) {
-    return(character())
-  }
   sidebar_section(
     heading = "Citation",
-    bullets = a(sprintf(tr_("Citing %s"), pkg$package), "authors.html")
+    bullets = a(sprintf(tr_("Citing %s"), pkg$package), "authors.html#citation")
```
Comment (maelle) [inline R/build-home-citation.R:36]: that ID wouldn't be translated, or it could, maybe, but it'd add complexity.

File: R/build-home-citation.R [hunk 2]
```diff
@@ -31,39 +31,55 @@ read_citation <- function(path = ".") {
...
-  source <- repo_source(pkg, "inst/CITATION")
+  source <- if (has_citation(pkg$src_path)) {
+    repo_source(pkg, "inst/CITATION")
+  } else {
+    repo_source(pkg, "DESCRIPTION")
+  }
 
   data <- list(
-    pagetitle = "Citation and Authors",
+    pagetitle = "Authors and Citation",
```
Comment (hadley) [inline R/build-home-citation.R:82]: This needs `tr_()` so it gets translated.

File: R/build-home-citation.R [hunk 3]
```diff
@@ -31,39 +31,55 @@ read_citation <- function(path = ".") {
...
+      cit <- list(
+        html = paste0("<p>", text_version, "</p>"),
+        bibtex = format(cit, style = "bibtex")
+      )
+    }
+
+    return(purrr::transpose(cit))
+  }
+
+  autocit <- utils::packageDescription(pkg$package)
```
Comment (hadley) [inline R/build-home-citation.R:63]: I'd pull this code out into a pair of functions like `citation_default()` or `citation_included()` (but with better names if you have any ideas)

File: inst/BS5/templates/content-citation-authors.html [hunk 4]
```diff
@@ -3,18 +3,9 @@
...
-    {{#citations}}
-    {{{html}}}
-    <pre>{{{bibtex}}}</pre>
-    {{/citations}}
 
-    <div class="page-header">
-      <h1>{{#translate}}{{authors}}{{/translate}}</h2>
     </div>
-   {{{before}}}
+     {{{before}}}
```
Comment (hadley) [inline inst/BS5/templates/content-citation-authors.html:8]: WS?

File: inst/BS5/templates/content-citation-authors.html [hunk 5]
```diff
@@ -24,6 +15,15 @@ <h1>{{#translate}}{{authors}}{{/translate}}</h2>
       {{/authors}}
     </ul>
    {{{after}}}
+
+    <div class="page-header">
```
Comment (hadley) [inline inst/BS5/templates/content-citation-authors.html:19]: This isn't a page header so it shouldn't get this class 😄

File: R/build-home-citation.R (hunk 1 above)
Comment (hadley) [reply R/build-home-citation.R:36]: Agreed; it's not user facing so I think it's fine as is.

File: R/build-home-citation.R [hunk 6]
```diff
@@ -31,39 +31,60 @@ read_citation <- function(path = ".") {
...
     ),
-    bibtex = format(cit, style = "bibtex")
+    bibtex = format(provided_citation, style = "bibtex")
   )
 
   purrr::transpose(cit)
 }
 
+citation_auto <- function(pkg) {
+  autocit <- utils::packageDescription(pkg$package)
```
Comment (hadley) [inline R/build-home-citation.R:68]: I think you can avoid the requirement to install the package by a little trickery: 

```R
autocit <- utils::packageDescription(path_file(pkg$src_path), lib.loc = path_dir(pkg$src_path))
```

File: inst/BS5/templates/content-citation-authors.html [hunk 7]
```diff
@@ -1,18 +1,10 @@
 <div class="row">
   <div class="contents col-md-9">
+    <section class="section level2 authors-section">
```
Comment (hadley) [inline inst/BS5/templates/content-citation-authors.html:3]: Yeah, I'd stick with divs for now, and we can consider if it's worth switching to `<sections>` later (and I doubt it's worth the effort)

File: R/build-home-citation.R (hunk 6 above)
Comment (maelle) [reply R/build-home-citation.R:68]: Cool trick! :sparkles:

**Comment by maelle** [comment]
I have added placeholders for "Authors and Citation" in .po files (or made the tweak when "Citation and Authors" was available and it was easy enough for me) but I do not know what else is needed.

**Comment by hadley** [comment]
I updated the translations by running `potools::po_extract()` (to get new messages from the R source code) and then `potools::po_update()` (to update the existing translations).

**Pull request merged**